### PR TITLE
[incubator/kafka] Update to StatefulSet & bump ZK dependency

### PR DESCRIPTION
**Pull request by krallistic**
Updated PetSet to StatefulSet.
Updated zookeeper chart version to newer 0.2.0 Version (contains affinity&pdb)

### CODE REVIEW

**Comment by mgoodness** [comment]
@k8s-bot ok to test

**Comment by mgoodness** [comment]
Resolves #630

File: incubator/kafka/requirements.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
 dependencies:
 - name: zookeeper
-  version: 0.1.1
+  version: 0.2.0
```
Comment (mgoodness) [inline incubator/kafka/requirements.yaml]: Latest version is 0.2.1

File: incubator/kafka/requirements.yaml [hunk 2]
```diff
@@ -1,4 +1,4 @@
...
   repository: http://storage.googleapis.com/kubernetes-charts-incubator
```
Comment (mgoodness) [inline incubator/kafka/requirements.yaml]: Please update to new URL: `https://kubernetes-charts-incubator.storage.googleapis.com`, update your cache (`helm repo add incubator https://kubernetes-charts-incubator.storage.googleapis.com`), then re-run `helm dep update incubator/kafka`

**Review by mgoodness — CHANGES_REQUESTED** [review]
Incubator repo URL needs to be changed in `requirements.yaml`. Let's also use the latest Zookeeper chart.

**Comment by krallistic** [comment]
Changed.

If i want to add struktural changes to the chart, should i open a second PR or use that one? (Also do i need a Issue? )

**Comment by prydonius** [comment]
@krallistic it's up to you - you're welcome to make more changes in this PR, but it might be easier to merge this as is and iterate on it with successive PRs.

**Comment by prydonius** [comment]
Waiting for @mgoodness to re-review. Can you bump the chart version?

**Comment by krallistic** [comment]
Thanks for the information.

Sadly i couldnt find out why the test is failing, my guess is that its because the kafka pods are failing till zookeeper is finally up. Maybe a init container wich waits for zK would be a good idea.

**Review by mgoodness — APPROVED** [review]

**Comment by mgoodness** [comment]
Changes look good, but we really should try to fix the failing test.

**Comment by viglesiasce** [comment]
@k8s-bot e2e test this

**Comment by viglesiasce** [comment]
This PR is actually creating a new file rather than updating the existing one. Please make your changes in the existing file templates/kafka-ss.yaml or move the file using git. 

As it stands this does not work. as the tests show.

**Comment by krallistic** [comment]
Also fixed in https://github.com/kubernetes/charts/pull/989, closing this PR

**Pull request closed without merging**